### PR TITLE
feat: [CDS-80416]: fix static schema for steps in custom stage

### PR DESCRIPTION
**Pull request by shalini-agr**

### CODE REVIEW

File: v1/pipeline/stages/custom/custom-stage-node.yaml [hunk 1]
```diff
@@ -42,5 +42,9 @@ allOf:
 anyOf:
   - required:
       - name
+      - type
```
Comment (archit-harness) [inline v1/pipeline/stages/custom/custom-stage-node.yaml:45]: id and name both are not required

File: v1/pipeline/stages/custom/custom-stage-spec-element-wrapper-config.yaml [hunk 2]
```diff
@@ -1,8 +1,6 @@
 title: CustomStageSpecElementWrapperConfig
 type: object
 properties:
-  parallel:
-    $ref: parallel-step-element-config.yaml
   step:
```
Comment (archit-harness) [inline v1/pipeline/stages/custom/custom-stage-spec-element-wrapper-config.yaml:4]: it doesnt contain step keyword also right?

File: v1/pipeline/steps/common/background-step-node.yaml [hunk 3]
```diff
@@ -1,56 +1,49 @@
...
-        const: Background
+        const: background
   then:
     properties:
       spec:
         $ref: ../common/background-step-info.yaml
+
+anyOf:
+  - required:
+      - name
```
Comment (archit-harness) [inline v1/pipeline/steps/common/background-step-node.yaml]: id and name both are not required

File: v1/pipeline/steps/common/run-step-node.yaml [hunk 4]
```diff
@@ -1,56 +1,49 @@
...
-        const: Run
+        const: run
   then:
     properties:
       spec:
         $ref: ../common/run-step-info.yaml
+
+anyOf:
+  - required:
+      - name
```
Comment (archit-harness) [inline v1/pipeline/steps/common/run-step-node.yaml]: same as above

File: v1/pipeline/steps/custom/container-step-node.yaml [hunk 5]
```diff
@@ -1,56 +1,48 @@
...
-        const: Container
+        const: container
   then:
     properties:
       spec:
         $ref: container-step-info.yaml
+anyOf:
+  - required:
+      - name
+      - type
```
Comment (archit-harness) [inline v1/pipeline/steps/custom/container-step-node.yaml]: same as above

File: v1/pipeline/steps/custom/wait-step-node.yaml [hunk 6]
```diff
@@ -1,53 +1,47 @@
...
       type:
-        const: Wait
+        const: wait
   then:
     properties:
       spec:
         $ref: wait-step-info.yaml
+
+anyOf:
+  - required:
```
Comment (archit-harness) [inline v1/pipeline/steps/custom/wait-step-node.yaml:39]: same as above

Comment (shalini-agr) [reply v1/pipeline/steps/custom/wait-step-node.yaml]: done

File: v1/pipeline/steps/custom/container-step-node.yaml (hunk 5 above)
Comment (shalini-agr) [reply v1/pipeline/steps/custom/container-step-node.yaml]: done

File: v1/pipeline/steps/common/run-step-node.yaml (hunk 4 above)
Comment (shalini-agr) [reply v1/pipeline/steps/common/run-step-node.yaml]: done

File: v1/pipeline/stages/custom/custom-stage-spec-element-wrapper-config.yaml (hunk 2 above)
Comment (shalini-agr) [reply v1/pipeline/stages/custom/custom-stage-spec-element-wrapper-config.yaml]: yes, removed step keyword

File: v1/pipeline/stages/custom/custom-stage-node.yaml (hunk 1 above)
Comment (shalini-agr) [reply v1/pipeline/stages/custom/custom-stage-node.yaml]: done

**Pull request closed without merging**